File: clients/ios/Lextures/Core/LMS/LMSFeatureModelsGamification.swift
import Foundation

// MARK: - Gamification (M9.3)

struct GamificationBadge: Codable, Identifiable, Hashable {
    var badgeType: String
    var awardedAt: String

    var id: String { "\(badgeType)-\(awardedAt)" }
}

struct GamificationProfile: Codable {
    var xpTotal: Int
    var level: Int
    var xpToNextLevel: Int
    var levelProgressPct: Double
    var currentStreak: Int
    var longestStreak: Int
    var streakFreezes: Int
    var streakAtRisk: Bool
    var streakHoursLeft: Double?
    var streakEnded: Bool?
    var leaderboardVisible: Bool
    var badges: [GamificationBadge]?
    var recentBadges: [GamificationBadge]?
}

struct LeaderboardEntry: Codable, Identifiable, Hashable {
    var rank: Int
    var userId: String
    var displayName: String
    var xpEarned: Int
    var isCurrentUser: Bool?

    var id: String { "\(rank)-\(userId)" }
}

struct CourseLeaderboardResponse: Codable {
    var topEntries: [LeaderboardEntry]?
    var currentUser: LeaderboardEntry?
}

struct GamificationBadgesListResponse: Codable {
    var badges: [GamificationBadge]?
}

File: clients/android/app/src/main/kotlin/com/lextures/android/core/lms/LmsFeatureModelsGamification.kt
package com.lextures.android.core.lms

import kotlinx.serialization.Serializable

@Serializable
data class GamificationBadge(
    val badgeType: String,
    val awardedAt: String,
)

@Serializable
data class GamificationProfile(
    val xpTotal: Int = 0,
    val level: Int = 0,
    val xpToNextLevel: Int = 0,
    val levelProgressPct: Double = 0.0,
    val currentStreak: Int = 0,
    val longestStreak: Int = 0,
    val streakFreezes: Int = 0,
    val streakAtRisk: Boolean = false,
    val streakHoursLeft: Double? = null,
    val streakEnded: Boolean? = null,
    val leaderboardVisible: Boolean = true,
    val badges: List<GamificationBadge>? = null,
    val recentBadges: List<GamificationBadge>? = null,
)

@Serializable
data class LeaderboardEntry(
    val rank: Int,
    val userId: String,
    val displayName: String,
    val xpEarned: Int,
    val isCurrentUser: Boolean? = null,
)

@Serializable
data class CourseLeaderboardResponse(
    val topEntries: List<LeaderboardEntry>? = null,
    val currentUser: LeaderboardEntry? = null,
)

@Serializable
data class GamificationBadgesListResponse(
    val badges: List<GamificationBadge>? = null,
)